pkg/mcp/sink/sink.go

```diff
@@ -194,6 +199,9 @@ func (sink *Sink) processStream(stream Stream) error {
 			req = initialRequests[0]
 			initialRequests = initialRequests[1:]
 		} else {
+			if err := sink.requestLimiter.Wait(stream.Context()); err != nil {
```

Comment:
This is rate limiting the sink receiving new configuration updates. If we did want client/sink-side rate limiting, I think it would be on the send path requesting new updates. 

For now I'd be inclined to focus more on the server/source request rate limiting per-connection.

Reply:
That seems reasonable, I was under assumption that there would be a rate limiter on both producing and consuming side. that way we can avoid fast producer and slow consumer problem in case if client side rate limiter is misbehaving due to bad config, etc.

Reply:
Both may be useful. I would move the consumer-side limit in the client to the send path so that we can always apply valid configuration ASAP. The delay would happen on the request for subsequent updates (e.g. ACK/NACK).

Reply:
Done.